### PR TITLE
misc: capture Semaphore PR number

### PR DESCRIPTION
**Pull request by estrada9166**
- Closes  [CF-1442](https://cypress-io.atlassian.net/browse/CF-1442)

### Additional details

Capture Semaphore PR number to correctly display it on the cloud

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

[CF-1442]: https://cypress-io.atlassian.net/browse/CF-1442?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jennifer-shehane — APPROVED** [review]
This env var is documented here: https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number

**Comment by jennifer-shehane** [comment]
@estrada9166 Can you add a changelog entry as outlined in this guide? https://github.com/cypress-io/cypress/blob/develop/guides/writing-the-cypress-changelog.md#writing-guidelines

**Review by AtofStryker — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+
+- We now capture the [Semaphore](https://semaphoreci.com/) CI provider's environment variable [`SEMAPHORE_GIT_PR_NUMBER`](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number) to display the linked PR number in the Cloud. Addressed in [#29314](https://github.com/cypress-io/cypress/pull/29314).
 
 _Released 4/11/2024_
 
+## 13.7.3
+
 **Bugfixes:**
 
 - Fixed an issue where asserts with custom messages weren't displaying properly. Fixes [#29167](https://github.com/cypress-io/cypress/issues/29167)
```
Comment (mschile) [inline cli/CHANGELOG.md]:
```suggestion
## 13.7.4

_Released 4/16/2024 (PENDING)_

**Misc:**

- We now capture the [Semaphore](https://semaphoreci.com/) CI provider's environment variable [`SEMAPHORE_GIT_PR_NUMBER`](https://docs.semaphoreci.com/ci-cd-environment/environment-variables/#semaphore_git_pr_number) to display the linked PR number in the Cloud. Addressed in [#29314](https://github.com/cypress-io/cypress/pull/29314).

## 13.7.3

_Released 4/11/2024_

**Bugfixes:**

- Fixed an issue where asserts with custom messages weren't displaying properly. Fixes [#29167](https://github.com/cypress-io/cypress/issues/29167)
```

**Review by mschile — APPROVED** [review]

**Comment by AtofStryker** [comment]
I wonder if next version is failing due to my entry of 13.8.0 in #29306r or if the semantic commit job thinks MISC is a feature bump? My guess is the later

**Review by jennifer-shehane — CHANGES_REQUESTED** [review]
@estrada9166 There's a failure in the server tests. 

Screenshot 2024-04-12 at 2 03 49 PM

**Comment by jennifer-shehane** [comment]
@AtofStryker Maybe it's failing because the initial commit was prefixed with `feat:`??

File: cli/CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 13.7.4
```
Comment (jennifer-shehane) [inline cli/CHANGELOG.md:2]:
```suggestion
## 13.8.0
```

**Review by jennifer-shehane — APPROVED** [review]

File: cli/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,12 @@
 <!-- See the ../guides/writing-the-cypress-changelog.md for details on writing the changelog. -->
+## 13.8.0
```
Comment (jennifer-shehane) [inline cli/CHANGELOG.md]:
```suggestion
## 13.7.4
```

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**